src/Razor/src/Microsoft.AspNetCore.Razor.Language/Intermediate/TagHelperPropertyIntermediateNode.cs

```diff
@@ -19,6 +19,8 @@ public sealed class TagHelperPropertyIntermediateNode : IntermediateNode
 
         public bool IsIndexerNameMatch { get; set; }
 
+        public bool IsDirectiveAttribute => BoundAttribute?.IsDirectiveAttribute() ?? false;
```

Comment:
New node

Reply:
I thought about it while doing this. I didn't do it because like you said it didn't seem important for the insertion. But I agree this would make things cleaner. I'm going to try this some time later and if it takes very long, I'll file an issue and merge this.

Reply:
Filed https://github.com/aspnet/AspNetCore/issues/10547.